### PR TITLE
Add GitHub Actions - docker.yaml

### PR DESCRIPTION
**Pull request by abraunegg**
* Add GitHub Actions - docker.yaml

### CODE REVIEW

**Comment by abraunegg** [comment]
@MarcelCoding
Revisiting your PR from #1733 .. managing to get a little further with the docker.yaml file - however it is now generating the following error - for all elements:
```
WARNING: No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
error: unable to prepare context: path "./onedrive" not found
Error: buildx failed with: error: unable to prepare context: path "./onedrive" not found
```

Capture


Any ideas?

**Comment by MarcelCoding** [comment]
The warning is only because you do not publish that image resulting from that build. The errors mean that the folder onedrive isn't there. If you build a Docker image, you have a context. That is a folder from where all you relative paths from the Dockerfile (COPY, ADD) are going to be resolved

**Comment by abraunegg** [comment]
@MarcelCoding 
So how to fix?

The .yaml is basically the same one from the original PR ... which I am guessing you tested before you submitted it originally ..

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if: github.event_name != 'pull_request'
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ github.token }}
+
+      - name: Build
+        uses: docker/build-push-action@v2
+        with:
+          context: ./onedrive
```
Comment (MarcelCoding) [inline .github/workflows/docker.yaml:81]: This line should be a ".". I tested the file, sorry for the inconvenience. I am only on my phone and did not saw that this is a pr. :)

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Comment by abraunegg** [comment]
@MarcelCoding 
OK .. that now gets things moving a little further .. but now all the builds fail with this or similar:
```
#23 3.660 fatal: No names found, cannot describe anything.
#23 3.661 make: *** [Makefile:101: onedrive] Error 128
#23 ERROR: process "/bin/sh -c autoreconf -fiv  && ./configure  && make clean  && make  && make install" did not complete successfully: exit code: 2
```

ideas?

**Comment by MarcelCoding** [comment]
That is now some kind of error from the make file. The lines stating with the decimel number are output of the real build and not from Docker

**Comment by abraunegg** [comment]
> That is now some kind of error from the make file. The lines stating with the decimel number are output of the real build and not from Docker

OK .. weird .. because - the Dockerfile's themselves - I can build without issue any of those containers in Docker manually - and run them - without issue ...

Anything in particular I should be looking for ?

**Comment by MarcelCoding** [comment]
Do you localay use `docker build` or `docker buildx build `? If the first one, could you try the second one. Another option would be some kind of a specific thing of a architecture.

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          - flavor: debian
+            dockerfile: ./contrib/docker/Dockerfile-debian
+            platforms: linux/amd64,linux/arm64,linux/arm/v7
+          - flavor: alpine
+            dockerfile: ./contrib/docker/Dockerfile-alpine
+            platforms: linux/amd64,linux/arm64
+
+    steps:
+      - name: Check out code from GitHub
+        run: git clone https://github.com/abraunegg/onedrive.git
```
[the remark this answers is not in the document]
Comment (MarcelCoding) [reply .github/workflows/docker.yaml:32]: https://github.com/actions/checkout

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (abraunegg) [reply .github/workflows/docker.yaml:32]: yes - I was just testing as that 'error' was coming from 'git'

**Comment by abraunegg** [comment]
@MarcelCoding 

> Do you localay use `docker build` or `docker buildx build `? If the first one, could you try the second one. Another option would be some kind of a specific thing of a architecture.


Locally I have tested:
* `docker build . -t local-ondrive-debian -f contrib/docker/Dockerfile-debian`
* `docker buildx build . -t local-ondrive-debian -f contrib/docker/Dockerfile-debian `

Both commands build the image without issue ... *puzzled*

**Comment by MarcelCoding** [comment]
Iam going to take a deeper look at this later today. Currently is school xD

File: .github/workflows/docker.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.flavor }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ matrix.flavor }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        if: github.event_name != 'pull_request'
```
Comment (MarcelCoding) [inline .github/workflows/docker.yaml:66]: .

File: .github/workflows/docker.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          username: ${{ github.repository_owner }}
+          password: ${{ github.token }}
+
+      - name: Build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ${{ matrix.dockerfile }}
+          platforms: ${{ matrix.platforms }}
+          push: ${{ github.event_name != 'pull_request' }}
```
Comment (MarcelCoding) [inline .github/workflows/docker.yaml:85]: .

**Review by MarcelCoding — COMMENTED** [review]
These tow things decide whether the should be performed a login and if the image should be pushed and because you are in a pr this is not going to be triggered. Additionally a pr from a fork does not has access to the secrets. Someone could modify the actions file and extract the secrets.

File: .github/workflows/docker.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        if: github.event_name != 'pull_request'
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        if: github.event_name != 'pull_request'
```
Comment (MarcelCoding) [inline .github/workflows/docker.yaml:73]: .

**Comment by abraunegg** [comment]
@MarcelCoding 

I have been working on #1978 testing ... and looks like I have finally got things to work ... double checking a number of things, then going to pull those changes here

**Comment by MarcelCoding** [comment]
That's great, although I don't see the thing that made fge difference. Only ghcr isn't there anymore?

**Comment by abraunegg** [comment]
@MarcelCoding 
OK .. this PR now looks good to go | merge ...

Question .. will this also now do all the 'latest' tags as per previous .. ?

File: .github/workflows/docker.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Docker meta
+        id: docker_meta
+        uses: marcelcoding/ghaction-docker-meta@v1
```
Comment (MarcelCoding) [inline .github/workflows/docker.yaml:39]: You could update this to v2. Yes for Ever release this will be tagged latest

Comment (abraunegg) [reply .github/workflows/docker.yaml:39]: > You could update this to v2

will do

**Comment by abraunegg** [comment]
@MarcelCoding 
Many thanks for your help .. all working well now.

**Comment by abraunegg** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**